core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -93,15 +91,15 @@ private[spark] class SparkUI private (
   /** Stop the server behind this web interface. Only valid after bind(). */
   override def stop() {
     super.stop()
-    logInfo("Stopped Spark web UI at %s".format(appUIAddress))
+    logInfo(s"Stopped Spark web UI at $appUIAddress")
   }
 
   /**
    * Return the application UI host:port. This does not include the scheme (http://).
    */
   private[spark] def appUIHostPort = publicHostName + ":" + boundPort
 
-  private[spark] def appUIAddress = s"http://$appUIHostPort"
+  private[spark] def appUIAddress = webUrl
```

Comment:
I think these two methods can just be eliminated, because appUIAddress does seem to return exactly the same thing, and appUIHostPort is only referenced from tests.